[BUG]: CommonJS TypeScript files do not find type declarations

### Description of the bug

When importing the used-pm package into a TypeScript file in CommonJS mode, `tsc` reports the following error:

```
foo.cts:1:20 - error TS7016: Could not find a declaration file for module 'used-pm'. '.../node_modules/used-pm/dist/index.cjs' implicitly has an 'any' type.
  If the 'used-pm' package actually exposes this module, try adding a new declaration (.d.ts) file containing `declare module 'used-pm';`

1 import usedPM from 'used-pm';
                     ~~~~~~~~~
```

### Steps To Reproduce

0. Go into the new directory and create a `package.json` file with the following contents:

    ```json
    {}
    ```

2. Run `npm install used-pm typescript`
3. Create a `tsconfig.json` file with the following contents:

    ```json
    {
      "compilerOptions": {
        "noImplicitAny": true,
        "moduleResolution": "node16"
      }
    }
    ```

4. Create a `foo.cts` file with the following contents:

    ```ts
    import usedPM from 'used-pm';
    console.log(usedPM());
    ```

5. Run `npx tsc --noEmit`

### Additional Information

If there is an `exports` field in `package.json`, TypeScript will now look for a type declaration file using the `"types"` import condition instead of the `types` field.
see https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing

Even if there is no `"types"` import condition, TypeScript will not use the `types` field.
So TypeScript **will not find a type declaration file**.

Therefore, the `exports` field should be written as follows:

```json
{
	"exports": {
		"types": "./dist/index.d.ts",
		"import": "./dist/index.js",
		"require": "./dist/index.cjs"
	}
}
```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "used-pm",
-	"version": "1.0.0",
+	"version": "1.0.1",
 	"description": "Detects what package manager executes the process",
 	"keywords": [
 		"npm",
@@ -13,16 +13,22 @@
 	"license": "MIT",
 	"author": "Alexander Böhm <tools@boehm.work>",
 	"exports": {
-		"import": "./dist/index.js",
-		"require": "./dist/index.cjs"
+		".": {
+			"import": {
+				"types": "./dist/esm/index.d.ts",
+				"default": "./dist/esm/index.js"
+			},
+			"require": {
+				"types": "./dist/cjs/index.d.ts",
+				"default": "./dist/cjs/index.js"
+			}
+		}
 	},
-	"types": "./dist/index.d.ts",
 	"files": [
 		"./dist/"
 	],
 	"scripts": {
-		"build": "tsup",
-		"dev": "tsup --watch",
+		"build": "tsc -b ./tsconfig.cjs.json ./tsconfig.esm.json",
 		"format": "prettier --write \"**/*.{cjs,js,mjs,ts,json,md,mdx,yaml,yml}\"",
 		"lint": "eslint . --ext .cjs,.js,.mjs,.ts --fix --ignore-path .gitignore",
 		"prepare": "ts-node ./scripts/prepare.ts && husky install"
@@ -49,21 +55,9 @@
 		"lint-staged": "^13.2.2",
 		"prettier": "^2.8.8",
 		"ts-node": "^10.9.1",
-		"tsup": "^6.7.0",
 		"typescript": "^5.0.4"
 	},
 	"publishConfig": {
 		"access": "public"
-	},
-	"tsup": {
-		"clean": true,
-		"entry": [
-			"src/index.ts"
-		],
-		"format": [
-			"cjs",
-			"esm"
-		],
-		"dts": true
 	}
 }

--- a/src/index.ts
+++ b/src/index.ts
@@ -3,7 +3,7 @@ interface PackageManager {
 	version: string;
 }
 
-export = (): PackageManager | undefined => {
+export default (): PackageManager | undefined => {
 	const userAgent = process.env.npm_config_user_agent;
 
 	if (!userAgent) return;

--- a/tsconfig.cjs.json
+++ b/tsconfig.cjs.json
@@ -1,0 +1,9 @@
+{
+	"extends": "./tsconfig.json",
+	"include": ["@types/**/*", "src/**/*"],
+	"compilerOptions": {
+		"outDir": "./dist/cjs",
+		"module": "CommonJS",
+		"declaration": true
+	}
+}

--- a/tsconfig.esm.json
+++ b/tsconfig.esm.json
@@ -1,0 +1,9 @@
+{
+	"extends": "./tsconfig.json",
+	"include": ["@types/**/*", "src/**/*"],
+	"compilerOptions": {
+		"outDir": "./dist/esm",
+		"module": "ESNext",
+		"declaration": true
+	}
+}

